### ISSUE TITLE
Audio doesn't regain focus after receiving Messenger call

### ISSUE THREAD

**Issue by kmrinmoy07** [google/ExoPlayer#8015]
### [REQUIRED] Issue description
I am using exoplayer to play audio files from my server. Everything works fine and I was able to handle audio focus correctly while other application starts playing sound. But accidentally yesterday I discovered that in case of Facebook's Messenger, after disconnecting an incoming call in Messenger, the audio doesn't play back automatically nor play back by hitting play button, even if I try to change the audio source by clicking on a different media still the changed audio does not plays back. But the track gets changed or gets skip or rewind, but the audio doesn't plays. May be the player does not gets the audio focus back. Only after I destroy the player and restart the player or after I reopen the app, the audio gets played.

But everything works fine when any other app except Messenger request audio focus.

This is how I am using my exoplayer in my app-

         AudioAttributes audioAttributes = new AudioAttributes.Builder()
            .setUsage(C.USAGE_MEDIA)
            .setContentType(C.CONTENT_TYPE_MOVIE)
            .build();
         exoPlayer = ExoPlayerFactory.newSimpleInstance(context);
         exoPlayer.addListener(listener);
         exoPlayer.setAudioAttributes(audioAttributes, true);

         ...
    
         DataSource.Factory dataSourceFactory = new DefaultDataSourceFactory(context,
                        Util.getUserAgent(context, context.getResources().getString(R.string.app_name)));
         audioSource =
                        new ProgressiveMediaSource.Factory(dataSourceFactory)
                                .setLoadErrorHandlingPolicy(new CustomPolicy())
                                .createMediaSource(Uri.parse(audio_url));
         exoPlayer.prepare(audioSource);
         exoPlayer.setPlayWhenReady(true);

     
      

### [REQUIRED] Reproduction steps
This can be reproduced just by getting an incoming voice call in Messenger and then receive the call and then after disconnecting you will be able to see the issue. But if you don't receive the call everything will be fine

### [REQUIRED] Link to test content
Any type of audio mp3 file can be used to test

### [REQUIRED] A full bug report captured from the device
Sorry, I was unable to get any stacktrace for this bug

### [REQUIRED] Version of ExoPlayer being used
The absolute version number is 2.11.1

### [REQUIRED] Device(s) and version(s) of Android being used
I am getting the issue on Redmi K20 Pro or also known as Xiaomi Mi 9T Pro and Android version is 10

**Comment by krocard** [comment]
This seems to be an audio focus problem. I'll try to reproduce.

**Comment by ojw28** [comment]
I think this is probably a bug in Messenger where they don't release transient audio focus. A similar issue was reported for Duo previously, in https://github.com/google/ExoPlayer/issues/7182.

There was also a bug in ExoPlayer where we'd not be aggressive enough at trying to take audio focus back in this case, which is why you observe that pressing the play button does not allow audio to resume, and that only recreating the player is sufficient. This issue is also described in the bug referenced above, and was fixed in 2.11.4. Hence upgrading to 2.11.4 or later should resolve this issue.

Marking as a duplicate of https://github.com/google/ExoPlayer/issues/7182.
